buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -285,7 +285,7 @@ final void ensureWritable0(int minWritableBytes) {
         final int writerIndex = writerIndex();
         final int targetCapacity = writerIndex + minWritableBytes;
         // using non-short-circuit & to reduce branching - this is a hot path and targetCapacity should rarely overflow
-        if (targetCapacity >= 0 & targetCapacity <= capacity()) {
+        if (targetCapacity >= 0 && targetCapacity <= capacity()) {
```

Comment:
There is a comment just above this line that explains why it is like this. It would be good to run some microbenchmarks to ensure we don't regress performance for the various buffer types.

Reply:
Got it, it confused me before, thanks a lot